packages/@aws-cdk/aws-ec2/lib/instance-types.ts

```diff
@@ -585,4 +606,18 @@ export class InstanceType {
   public toString(): string {
     return this.instanceTypeIdentifier;
   }
+
+  /**
+   * Returns the CPU architecture for the instance type
+   */
+  private resolveArchitecture(): InstanceArchitecture {
+    const instanceClass = this.instanceTypeIdentifier.split('.')[0];
+
+    // Graviton-based (a1, t4g, c6g, c6gd, c6gn, m6g, m6gd, r6g, r6gd)
+    if (/^a1|[cmrt][\d]g[dn]?$/.test(instanceClass)) {
```

Comment:
I like doing it by regex, but I think we should simplify the regex some so we can automatically account for future instance  types (otherwise this will be chasing a moving target).

An ARM instance is identified by either:

* Starts with an `a` for instance family ARM; or
* Has a `g` in the "extra capabilities" for Graviton

Also we're getting dangerously close to generation 10 so I'd like to be safe against double-digit generations.


I'll propose:

```
/^a|^[^.]*\d[^.]*g[^.]*\./
```

In other words: starts with an `a` OR has a `g` somewhere after a digit before the first period.

Probably worth writing down what it's testing for in a comment next to it :)



![image](https://user-images.githubusercontent.com/524162/103769921-71e86600-5025-11eb-89c5-641af9943641.png)

Reply:
Thank you for the feedback, I have attempted to improve the regex and make it more future-proof

Reply:
I refined it yet again. Now we just match against the whole instanceTypeIdentifier and capture the family-part and capabilities-part and test against those instead.